### PR TITLE
Level up markdown content.

### PR DESCRIPTION
**Pull request by dblock**
Signed-off-by: dblock <dblock@amazon.com>

### Description

Coming from https://github.com/opensearch-project/.github/issues/8, use standard markdown content. 

### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by peterzhuamazon** [comment]
Is this standard apply to all the repos within opensearch-project now? @dblock

**Comment by dblock** [comment]
> Is this standard apply to all the repos within opensearch-project now? @dblock

Yes, see https://github.com/opensearch-project/.github/issues/8. Particularly important are things like copyright notices that seem to be incorrect in this repo rn.

File: MAINTAINERS.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+## Maintainers
+
+| Maintainer | GitHub ID | Affiliation |
+| --------------- | --------- | ----------- |
+| Peter Zhu | [peterzhuamazon](https://github.com/peterzhuamazon) | Amazon |
+| Peter Nied | [peternied](https://github.com/peternied) | Amazon |
+| Cameron Skinner | [camerski](https://github.com/camerski) | Amazon |
+| Barani Bikshandi | [bbarani](https://github.com/bbarani) | Amazon |
```
Comment (gaiksaya) [inline MAINTAINERS.md:8]: Please add me :) https://github.com/gaiksaya/

Comment (dblock) [reply MAINTAINERS.md:8]: Done!

**Review by peterzhuamazon — APPROVED** [review]

**Review by bbarani — APPROVED** [review]

**Pull request merged**